source/service.go

```diff
@@ -439,6 +439,14 @@ func (sc *serviceSource) generateEndpoints(svc *v1.Service, hostname string, pro
 		DNSName:    hostname,
 	}
 
+	eaAAAA := &endpoint.Endpoint{
+		RecordTTL:  ttl,
+		RecordType: endpoint.RecordTypeAAAA,
+		Labels:     endpoint.NewLabels(),
+		Targets:    make(endpoint.Targets, 0, defaultTargetsCapacity),
+		DNSName:    hostname,
+	}
```

Comment:
Where are the tests for the changes to this file?

Reply:
I'm not exactly sure what kind of tests would this need?

Reply:
There should be tests for both IPv6-only and dual-stack services.

Reply:
I believe I should have those now.